include/bluetooth/gatt.h

```diff
@@ -1344,8 +1343,16 @@ struct bt_gatt_read_params {
 			/** Attribute data offset. */
 			uint16_t offset;
 		} single;
-		/** Handles to read in Read Multiple Characteristic Values. */
-		uint16_t *handles;
+		struct {
+			/** Attribute handles to read with Read Multiple
+			 *  Characteristic Values.
+			 */
+			uint16_t *handles;
+			/** If true, we expect that the read characteristics
+			 *  have different lengths
+			 */
+			bool variable;
+		} multi;
```

Comment:
So maybe I just should add duplicate of `bt_gatt_read`, named maybe `bt_gatt_read`, that would call just `gatt_read_mult`,  and build it just for testing, so guard it with `#if defined(CONFIG_BT_TESTING)`? That way it wouldn't change the general API

Reply:
Having test-only code wouldn't make much sense. It would make sense to allow the application to use the most efficient read operation needed (related to the ATT_MTU and response length). So I think exposing it at the application level makes sense.
Doing the API change would be the proper solution.